cedar-policy-cli/src/lib.rs

```diff
@@ -104,8 +143,10 @@ impl RequestArgs {
         match &self.request_json_file {
             Some(jsonfile) => {
                 let jsonstring = std::fs::read_to_string(jsonfile)
-                    .context(format!("failed to open request-json file {jsonfile}"))?;
+                    .into_diagnostic()
+                    .wrap_err_with(|| format!("failed to open request-json file {jsonfile}"))?;
                 let qjson: RequestJSON = serde_json::from_str(&jsonstring)
+                    .into_diagnostic()
                     .context(format!("failed to parse request-json file {jsonfile}"))?;
```

Comment:
Oops, thanks for spotting this! `wrap_err` (and `wrap_err_with`) is more idiomatic for `miette`. `context` and (`context_with`) is defined as an alias for porting code over from a`anyhow`. I originally had code left using the `context` wrapper when I had compatibility things in place for SemVer reasons, and missed this when ripping them out.

Additionally, `wrap_err_with` (`context_with` for `anyhow`) should be used here because `format!` allocates a new string. Using the `_with` variant and wrapping that up in a lambda means the allocation only happens on the error path.